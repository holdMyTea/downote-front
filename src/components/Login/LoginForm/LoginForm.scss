@import '../../../style.scss';

.login-form {
  @extend %center-wrapper;
}

.login-form-group {
  display: flex;
  flex-direction: column;
}

.login-label {
  margin: 5px 0;
  font-family: $regular-font;
  font-size: $login-input-font-size;
}

%login-form-element {
  font-family: $regular-font;
  font-size: $login-input-font-size;
  border-radius: 5px;
}
.login-input {
  @extend %login-form-element;
  min-width: $login-field-min-width;
  margin-bottom: 15px;
  height: 35px;
}
.login-button {
  @extend %login-form-element;
  padding: 5px 10px;
}

.input-required {
  outline:none;
  border-color: $color-input-pending;
  box-shadow:0 0 10px $color-input-pending;
}
.input-wrong {
  outline:none;
  border-color: $color-input-wrong;
  box-shadow:0 0 10px $color-input-wrong;
}
.input-correct {
  outline:none;
  border-color: $color-input-correct;
  box-shadow:0 0 10px $color-input-correct;
}

%smooth-transition{
  transition: background-color 0.5s ease;
  transition: border 0.5s ease;
  transition: color 0.5s ease;
}
.button-pending {
  @extend %smooth-transition;
  border: 2px solid $color-secondary;
  background-color: white;
  color: black;
}
.button-ready {
  @extend %smooth-transition;
  border: 2px solid $color-input-correct;
  background-color: $color-input-correct;
  color: white;
}
