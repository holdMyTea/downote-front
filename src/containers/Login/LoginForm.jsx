import React, { useState } from 'react'
import Types from 'prop-types'
import { Form } from 'semantic-ui-react'

/**
 * Component with two inputs, email and password, and submit button
 * @param {string} [email=''] value to pre-fill email input with
 * @param {function} onSubmit submit function will be called with email string as the first param and password as the second one
 */
const LoginForm = ({ email = '', onSubmit }) => {
  // stores the current email input value
  const [ emailInput, setEmailInput ] = useState(email)
  // used for applying styles to email input and submit button
  const [ isEmailValid, setEmailValid ] = useState(true)

  // stores the current pass input value
  const [ passInput, setPassInput ] = useState('')
  // used for applying styles to pass input and submit button
  const [ isPassValid, setPassValid ] = useState(true)

  // validates email input
  const onEmailChange = (event) => {
    const newInput = event.target.value
    setEmailInput(newInput)
    setEmailValid(newInput.length === 0 ||
      /(\w)+@(\w)+\.{1}\w{1,5}/.test(newInput))
  }

  // validates pass input
  const onPassChange = (event) => {
    const newInput = event.target.value
    setPassInput(newInput)
    setPassValid(
      newInput.length === 0 || newInput.length > 5
    )
  }

  // checking for inputs' validity and lentgth to disable the submit button
  const isButtonDisabled = !(isEmailValid && isPassValid) ||
      emailInput.length === 0 || passInput.length === 0

  const handleSubmit = (event) => {
    event.preventDefault()
    onSubmit(emailInput, passInput)
  }

  return (
    <Form>
      <Form.Input
        label='Email'
        type='email'
        placeholder='Email'
        error={!isEmailValid}
        onChange={onEmailChange} />

      <Form.Input
        label='Password'
        type='password'
        placeholder='Password'
        error={!isPassValid}
        onChange={onPassChange} />

      <Form.Button
        fluid
        onClick={handleSubmit}
        disabled={isButtonDisabled}
        positive={!isButtonDisabled}
      >
        Log In
      </Form.Button>
    </Form>
  )
}

LoginForm.propTypes = {
  email: Types.string,
  onSubmit: Types.func.isRequired
}

export default LoginForm
