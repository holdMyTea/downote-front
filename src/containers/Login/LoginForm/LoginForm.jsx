import React, { Component } from 'react'
import Types from 'prop-types'

import './LoginForm.scss'

class LoginForm extends Component {
  constructor (props) {
    super(props)

    this.INPUT_REQUIRED = 'input-required'
    this.INPUT_CORRECT = 'input-correct'
    this.INPUT_WRONG = 'input-wrong'

    this.state = {
      emailValue: props.email || '',
      emailStatus: this.INPUT_REQUIRED,
      passValue: '',
      passStatus: this.INPUT_REQUIRED
    }
  }

  render () {
    return (
      <form className='login-form'>

        <div className='login-form-group'>
          <label className='login-label'>
            Email:
          </label>
          <input
            className={`login-input ${this.state.emailStatus}`}
            type='text' value={this.state.emailValue}
            onChange={this.onEmailChange} placeholder={'Email'}/>
        </div>

        <div className='login-form-group'>
          <label className='login-label'>
            Password:
          </label>
          <input
            className={`login-input ${this.state.passStatus}`}
            type='password' value={this.state.passValue}
            onChange={this.onPassChange} placeholder={'Password'}/>
        </div>

        <button className={`login-button ${this.updateButtonStyle()}`}
          onClick={this.handleSubmit}>
          Log In
        </button>

      </form>
    )
  }

  onEmailChange = (event) => {
    const newInput = event.target.value
    this.setState({
      emailValue: newInput,
      emailStatus: newInput.length === 0 ? this.INPUT_REQUIRED
        : /(\w)+@(\w)+\.{1}\w{1,5}/.test(newInput) ? this.INPUT_CORRECT
          : this.INPUT_WRONG
    })
  }

  onPassChange = (event) => {
    const newInput = event.target.value
    this.setState({
      passValue: newInput,
      passStatus: newInput.length === 0 ? this.INPUT_REQUIRED
        : newInput.length > 5 ? this.INPUT_CORRECT
          : this.INPUT_WRONG
    })
  }

  updateButtonStyle = () => {
    const s = this.state
    return ((s.emailStatus === this.INPUT_CORRECT) &&
      (s.passStatus === this.INPUT_CORRECT))
      ? 'button-ready' : 'button-pending'
  }

  handleSubmit = (event) => {
    event.preventDefault()
    this.props.onLogin(
      this.state.emailValue,
      this.state.passValue
    )
  }
}

LoginForm.propTypes = {
  email: Types.string,
  onLogin: Types.func.isRequired
}

export default LoginForm
