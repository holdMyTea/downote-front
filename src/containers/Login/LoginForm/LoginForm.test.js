import React from 'react'
import renderer from 'react-test-renderer'

import LoginForm from './LoginForm'

test('Note renders correctly', () => {
  const component = renderer.create(
    <LoginForm
      email=''
      onSubmit={() => ({})}
    />
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
