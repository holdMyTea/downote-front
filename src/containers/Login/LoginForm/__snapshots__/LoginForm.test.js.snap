// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note renders correctly 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label>
      Email
    </label>
    <div
      className="ui input"
    >
      <input
        onChange={[Function]}
        placeholder="Email"
        type="email"
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label>
      Password
    </label>
    <div
      className="ui input"
    >
      <input
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
  </div>
  <div
    className="disabled field"
  >
    <button
      className="ui fluid disabled button"
      disabled={true}
      onClick={[Function]}
      tabIndex={-1}
    >
      Log In
    </button>
  </div>
</form>
`;
